src/pages/List.js

```diff
@@ -5,18 +5,47 @@ export default function List(props) {
   const [listItem, loading, error] = useCollection(db.collection(props.token), {
     snapshotListenOptions: { includeMetadataChanges: true },
   });
+
+  const markItemPurchased = (e, id) => {
+    const elapsedMilliseconds = Date.now();
+
+    if (e.target.checked === true) {
+      db.collection(props.token).doc(id).update({
+        last_purchased: elapsedMilliseconds,
+      });
+    } else {
+      db.collection(props.token).doc(id).update({
+        last_purchased: null,
+      });
+    }
+  };
+
+  function compareTimeStamps(lastPurchased) {
+    const currentElapsedMilliseconds = Date.now();
+    const millisecondsInOneDay = 86400000;
+    return currentElapsedMilliseconds - lastPurchased < millisecondsInOneDay;
+  }
+
   return (
     <>
-      <h1>THIS IS THE LIST</h1>
-      <h2>Your token: {props.token}</h2>
+      <h1>Your token: {props.token}</h1>
       {error && <strong>Error: {JSON.stringify(error)}</strong>}
       {loading && <span>Collection: Loading...</span>}
       {listItem && (
         <>
-          <span>Collection:</span>
+          <span>Your Shopping List:</span>
           <ul>
             {listItem.docs.map((doc) => (
-              <li key={doc.id}>{JSON.stringify(doc.data())}</li>
+              <div className="checkbox-wrapper">
+                <input
+                  type="checkbox"
+                  defaultChecked={
+                    compareTimeStamps(doc.data().last_purchased) ? true : false
+                  }
+                  onClick={(e) => markItemPurchased(e, doc.id)}
+                />
+                <li key={doc.id}>{doc.data().item_name}</li>
+              </div>
```

Comment:
React is showing a warning in console because your key on `<li>` is not the immediate child of `<ul>`. 
```suggestion
              <li key={doc.id} className="checkbox-wrapper">
                <label>
                  <input
                    type="checkbox"
                    defaultChecked={compareTimeStamps(doc.data().last_purchased)}
                    onClick={(e) => markItemPurchased(e, doc.id)}
                  />
                  {doc.data().item_name}
                </label>
              </li>
```
Because your `compareTimeStamps(doc.data().last_purchased)' returns a boolean, you don't need to do a ternary here.

[More info](https://a11y-101.com/development/checkboxes) on using `label` to make forms accessible.

Reply:
Thanks for the suggestions @yenly! I have updated the `<li>` to be a direct child of `<ul>`.

I removed the ternary from the `defaultChecked` property (that's way nicer to look at!)

And added a `<label>` around the `<input />`.